docs-dev/DEVELOPMENT_LOCAL.md

```diff
@@ -0,0 +1,102 @@
+
+```
+daemon --port 7007 --hostname 0.0.0.0 --network dev-unstable --anchor-service-api https://cas-dev.3boxlabs.com --debug true --ethereum-rpc https://rinkeby.infura.io/v3/b6685df41e1647c4be0046dfa62a020b
+```
+This configuration will allow running the ceramic daemon from the debugger and setting breakpoints in the daemon.
+
+
+## Local Development
+
+CLI helpers commands on how to start up environment and debug local code base.  
+
+*Compile full library:*
+
+`npm run clean && npm install && npm run build`
+
+*How to run locally compiled ceramic code:*
+
+`cd ~/js-ceramic/packages/cli/bin`
+
+`node ceramic.js daemon`
+
+*Run docker with local code base:*
+
+`docker build -t js-ceramic-debug::latest -f Dockerfile.daemon .`
+
+*Clayground (full environment containerized - https://github.com/ceramicnetwork/clayground)*
+
+`docker run --rm --name ceramic -p 7007:7007 ceramicnetwork/js-ceramic:dev daemon --port "7007" --hostname 0.0.0.0 --network dev-unstable --anchor-service-api https://cas-dev.3boxlabs.com --debug true --ethereum-rpc https://rinkeby.infura.io/v3/b6685df41e1647c4be0046dfa62a020b`
+
+
+## Postman Collection
+
+Import this [Postman Collection](postman_collection.json) of queries to see a sample of API calls that can be used to exercise and explore common endpoints.
+
+
+## Packages Overview
+
+`3id-did-resolver` - 3Box ID
+`key-did-resolver` - blockchain public keys
+
+`blockchain-utils-*`
+
+* `linking` - client side app devs (that would like to auth ceramic using specific blockchains / cosmos, eth, etc.)
+
+* `validation` - contract part - eth accounts, against blockchain
+
+responsible for creation and validation. Once we link an account you have to request the signature from your blockchain.
+Responsible
+
+etherum.ts/createLink(did) -> sign
+
+`canary-integration` - standalone test (capability, tile & tile update)
+
+`cli` - package for the ceramic cli (mostly daemon) -> see examples at the beginning how to start it up
+`common` - helper functions
+
+`core` - code for ceramic node (invoked via ceramic daemon) // internal http and core API
+* `store` - pinning and level state store; snapshot of a stream state
+* `state-management` - stream is mutable; check on state progression. Some streams are better kept in memory vs. some street could be offloaded to storage
+* `pubsub` - node to node communication - push based messaging; Mostly implemented as Observable
+* `anchor` - anchoring or validating it inside the ceramic node (
+
+`http-client` - with any remote node - http endpoint -> connect to a node
+
+`ipfs-daemon` - DEPRECATED
+
+`ipfs-topology` - Bootstrap Mechanism
+
+`pinning-*` - all for pinning
+
+* `powergate` - 3BoxLabs integration with 3rd party SPI for $FILECOIN support (worked at some point but not sure anymore)
+
+* `crust` - crust.network - alternative to filecoin
+
+* `aggregation` -> use your own IPFS node
```

Comment:
not sure how this works, link?

Reply:
Makes multiple pinning backends appear as one. Allows one to pin data to multiple backends (IPFS, Arweave, Crus, Filecoin,...) simultaneously.

Reply:
I've added @ukstv explanation next to aggregation.